docs/source/private-data/private-data.md

```diff
@@ -347,6 +348,346 @@ submitted a letter of credit, prior to the execution of the transfer chaincode.
+
+### Deploy the test network
+
+We need to deploy an instance of the Fabric test network to run the smart contract. Open a command terminal and navigate to test network directory in your local clone of [fabric-samples](https://github.com/hyperledger/fabric-samples). We will operate from the `test-network` directory for the remainder of the tutorial.
+```
+cd fabric-samples/test-network
+```
+
+You can then deploy the network with the following command:
+```
+./network.sh up createChannel
+```
+The script will deploy the nodes of the network create a single channel named `mychannel` with Org1 and Org2 as channel members. We will use this channel to deploy the smart contract and trade our asset.
+
+#### Set the environment variables to operate as Org1
+
+In the course of running this sample, you need to interact with the network as both Org1 and Org2. To make the tutorial easier to use, we will use separate terminals for each organization. Open a new terminal and make sure that you are operating from the `test-network` directory. Set the following environment variables to operate the `peer` CLI as the Org1 admin:
+```
+export PATH=${PWD}/../bin:${PWD}:$PATH
+export FABRIC_CFG_PATH=$PWD/../config/
+export CORE_PEER_TLS_ENABLED=true
+export CORE_PEER_LOCALMSPID="Org1MSP"
+export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org1.example.com/users/Admin@org1.example.com/msp
+export CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
+export CORE_PEER_ADDRESS=localhost:7051
+```
+The environment variables also specify the endpoint information of the Org1 peer to submit requests.
+
+#### Set the environment variables to operate as Org2
+
+Now that we have one terminal that we can operate as Org1, open a new terminal for Org2. Make sure that this terminal is also operating from the `test-network` directory. Set the following environment variables to operate as the Org2 admin:
+```
+export PATH=${PWD}/../bin:${PWD}:$PATH
+export FABRIC_CFG_PATH=$PWD/../config/
+export CORE_PEER_TLS_ENABLED=true
+export CORE_PEER_LOCALMSPID="Org2MSP"
+export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org2.example.com/users/Admin@org2.example.com/msp
+export CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt
+export CORE_PEER_ADDRESS=localhost:9051
+```
+
+You will need switch between the two terminals as you go through the tutorial.
+
+### Deploy the chaincode
+
+Now that we can operate as both organizations, we need install the private asset transfer smart contract on the peers of Org1 and Org2, and deploy the chaincode to the channel approving and committing the chaincode definition.
+
+#### Install and approve the chaincode as Org1
+
+Open the Org1 terminal. Run the following command to package the private asset transfer chaincode:
```

Comment:
for golang, before package, I think we should add that:

package your chaincode dependencies using:
```
go mod tidy
go mod vendor
```

Reply:
I agree tidying is a good practice but I don't believe it's a useful addition to the sample. On `go mod vendor`, why? This sample does not have a vendor folder.

Reply:
I take mod vendor back. For golang , deployCC,sh does call that to package the go dependency in vendor folder. So no need.

Reply:
In 2.x it doesn't matter. If a `go.mod` is present the Peer will explicitly enter module mode at compilation for you. If a `vendor` directory is present the peer will explicitly disable module mode.

Vendoring is useful in places where the internet is not necessarily particularly stable, for example China/India. It's not uncommon to get timeouts when installing chaincode when the dependencies aren't included as part of the vendor folder and Go has to fetch them. This is a frequent complaint from community members outside North America.